smart_open/tests/test_smart_open.py

```diff
@@ -1163,6 +1163,19 @@ def test_write_text_gzip(self):
             actual = fin.read()
         self.assertEqual(text, actual)
 
+class HostNameTest(unittest.TestCase):
+    def test_host_name(self):
+        host = 'http://a.com/b'
+        expected = 'http://a.com/b'
+        self.assertTrue(expected == smart_open_lib._add_sheme_to_host(host))
+        host = 'a.com/b'
```

Comment:
Each of these should be a separate test case. Putting multiple checks into a single test case is bad practice, because one failure can mask subsequent ones.